python/coinbase-agentkit/coinbase_agentkit/wallet_providers/cdp_evm_server_wallet_provider.py

```diff
@@ -0,0 +1,344 @@
+from pydantic import BaseModel, Field
+from web3 import Web3
+from web3.types import BlockIdentifier, ChecksumAddress, HexStr, TxParams
+
+from ..network import NETWORK_ID_TO_CHAIN, Network
+from .evm_wallet_provider import EvmWalletProvider
+
+
+class CdpEvmServerProviderConfig(BaseModel):
+    """Configuration options for CDP EVM Server providers."""
+
+    api_key_id: str | None = Field(None, description="The CDP API key ID")
+    api_key_secret: str | None = Field(None, description="The CDP API secret")
+    wallet_secret: str | None = Field(None, description="The CDP wallet secret")
+
+
+class CdpEvmServerWalletProviderConfig(CdpEvmServerProviderConfig):
+    """Configuration options for CDP EVM Server wallet provider."""
+
+    network_id: str | None = Field(None, description="The network id")
+    address: str | None = Field(None, description="The address to use")
+    idempotency_key: str | None = Field(None, description="The idempotency key for wallet creation")
+
+
+class CdpEvmServerWalletProvider(EvmWalletProvider):
+    """A wallet provider that uses the CDP EVM Server SDK."""
+
+    def __init__(self, config: CdpEvmServerWalletProviderConfig):
+        """Initialize CDP EVM Server wallet provider.
+
+        Args:
+            config (CdpEvmServerWalletProviderConfig | None): Configuration options for the CDP provider. If not provided,
+                   will attempt to configure from environment variables.
+
+        Raises:
+            ValueError: If required configuration is missing or initialization fails
+
+        """
+        try:
+            self._api_key_id = config.api_key_id or os.getenv("CDP_API_KEY_ID")
+            self._api_key_secret = config.api_key_secret or os.getenv("CDP_API_KEY_SECRET")
+            self._wallet_secret = config.wallet_secret or os.getenv("CDP_WALLET_SECRET")
+
+            if not self._api_key_id or not self._api_key_secret or not self._wallet_secret:
+                raise ValueError(
+                    "Missing required environment variables. CDP_API_KEY_ID, CDP_API_KEY_SECRET, CDP_WALLET_SECRET are required."
+                )
+
+            network_id = config.network_id or os.getenv("NETWORK_ID", "base-sepolia")
+            self._idempotency_key = config.idempotency_key or os.getenv("IDEMPOTENCY_KEY")
+
+            chain = NETWORK_ID_TO_CHAIN[network_id]
+            rpc_url = chain.rpc_urls["default"].http[0]
+
+            self._network = Network(
+                protocol_family="evm",
+                network_id=network_id,
+                chain_id=chain.id,
+            )
+            self._web3 = Web3(Web3.HTTPProvider(rpc_url))
+
+            client = self.get_client()
+            if config.address:
+                account = asyncio.run(self._get_account(client, config.address))
+            else:
+                account = asyncio.run(self._create_account(client))
+
+            self._address = account.address
+            self._account = account
```

Comment:
nit: you could probably reduce the state to just `self._account`, and get the address from `self._account.address` wherever you need it

Reply:
Checked and there were no references to `self._account`, just `self._address`, so removed that instead